README-mux.md

```diff
@@ -0,0 +1,327 @@
+
+Check `/tmp/ansible.log` if there are any errors during the run.  If this
+hangs, just kill it and run it again - Ansible is (mostly) idempotent.  Same
+applies if there are any errors during the run - fix the machine and/or the
+`vars.yaml` and run it again.
+
+To confirm that OpenShift and logging are working:
+
+    # oc project logging
+    # oc get pods
+
+You should see the Elasticsearch, Curator, Kibana, and Fluentd pods running.
+
+## Running mux
+
+You will need a publicly accessible hostname to access the mux.  You can use a
+hostname alias in `/etc/hosts` as described above in
+[Installing](#installing-viaq)
+
+    10.16.19.171 openshift.logging.test kibana.logging.test mux.logging.test
+
+Download and run the [setup-mux.sh](setup-mux.sh) script:
+
+    # oc project logging
+    # curl https://raw.githubusercontent.com/ViaQ/Main/master/setup-mux.sh > setup-mux.sh
+    # chmod +x setup-mux.sh
+    # MUX_HOST=mux.logging.test ./setup-mux.sh
+
+This will create the `logging-mux` configmap, secrets, dc, pod, and service.
+
+    # oc get pods -l component=mux
+    NAME                  READY     STATUS    RESTARTS   AGE
+    logging-mux-1-yxx1o   1/1       Running   0          1h
+    # oc get svc logging-mux
+    NAME          CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE
+    logging-mux   172.30.215.88   10.0.0.3      24284/TCP   1h
+
+The externalIP value should be the IP address of your `eth0` interface, or
+whichever interface on the machine is used for external egress.  The script
+should automatically determine this.  If it is unable to, specify it with
+the environment variable `MUX_PUBLIC_IP`:
+
+    # MUX_HOST=mux.logging.test MUX_PUBLIC_IP=192.168.122.4 ./setup-mux.sh
+
+You can test with `openssl s_client` like this:
+
+    # echo hello | openssl s_client -quiet -connect localhost:24284
+    depth=1 CN = openshift-signer@1480618407
+    verify return:1
+    depth=0 CN = mux
+    verify return:1
+    ��HELO��nonce�Aa���3�/��Ps��auth��keepaliveÕ�PONG´invalid ping message��
+
+This means you are able to access the Fluentd secure_forward listener.  Don't
+worry about the garbage characters and invalid message.
+
+There is a test script [test-mux.sh](test-mux.sh) you can use to test that the
+mux is working.  It will reconfigure the regular OpenShift Fluentd to send its
+logs to the mux instead of to Elasticsearch directly, and then add some log
+messages to see if the mux sends them to Elasticsearch.
+
+    # oc project logging
+    # curl https://raw.githubusercontent.com/ViaQ/Main/master/test-mux.sh > test-mux.sh
+    # chmod +x test-mux.sh
+    # MUX_HOST=mux.logging.test ./test-mux.sh
+
+To search Elasticsearch, first get the name of the Elasticsearch pod:
+
+    # oc project logging
+    # espod=`oc get pods -l component=es -o jsonpath='{.items[0].metadata.name}'`
+
+Then use `oc exec` and `curl` like this:
+
+    # oc exec $espod -- curl --connect-timeout 1 -s -k \
```

Comment:
It is quoted as preformatted.  If you look at it with the github markdown renderer you can see that it is displayed as code/preformatted: https://github.com/richm/Main/blob/495a4d48c545802a152d9ff00677254ea99d80d6/README-mux.md